components/script/dom/document.rs

```diff
@@ -363,8 +363,12 @@ impl Document {
     }
 
     // https://dom.spec.whatwg.org/#concept-document-url
-    pub fn url(&self) -> &Url {
-        &self.url
+    pub fn url(&self) -> Ref<Url> {
+        self.url.borrow()
+    }
+
+    pub fn set_url(&self, url: &Url) {
+        *self.url.borrow_mut() = url.clone();
```

Comment:
Why is a `DOMRefCell` necessary? Does returning a mutable reference to `self.url` not work?

Also, we need to leave a FIXME here pointing to #13437.